geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -323,28 +361,67 @@ public String toString() {
     return null;
   }
 
-  private double processByteArrayAsDouble(byte[] value) {
-    String stringValue = bytesToString(value).toLowerCase();
-    double processedDouble;
-    switch (stringValue) {
-      case "inf":
-      case "+inf":
-      case "infinity":
-      case "+infinity":
-        processedDouble = POSITIVE_INFINITY;
-        break;
-      case "-inf":
-      case "-infinity":
-        processedDouble = NEGATIVE_INFINITY;
-        break;
-      default:
-        try {
-          processedDouble = Double.parseDouble(stringValue);
-        } catch (NumberFormatException nfe) {
-          throw new NumberFormatException(ERROR_NOT_A_VALID_FLOAT);
-        }
-        break;
+  private static double processByteArrayAsDouble(byte[] value) {
+    if (isPositiveInfinity(value)) {
+      return POSITIVE_INFINITY;
+    }
+    if (isNegativeInfinity(value)) {
+      return NEGATIVE_INFINITY;
+    }
```

Comment:
Would it make sense to move this implementation into `Coder.bytesToDouble()` so that it's available everywhere?